game/data/weapons.py

```diff
@@ -53,8 +53,9 @@ def name(self) -> str:
     def __setstate__(self, state: dict[str, Any]) -> None:
         # Update any existing models with new data on load.
         updated = Weapon.with_clsid(state["clsid"])
-        state.update(updated.__dict__)
-        self.__dict__.update(state)
+        if updated is not None:
```

Comment:
This needs a comment. I'm not really sure what the goal is.

Reply:
The goal is to prevent `updated.__dict__` from generating an exception if `updated is None`. Remember, `Weapon.with_clsid` can now return `None` which is used to identify incompatible (for whatever reason) weapons...